packages/frontend-api/src/Model/Category/CategoryRepository.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Category;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+use Shopsys\FrameworkBundle\Component\String\DatabaseSearching;
+use Shopsys\FrameworkBundle\Model\Category\Category;
+use Shopsys\FrameworkBundle\Model\Category\CategoryDomain;
+
+class CategoryRepository
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected EntityManagerInterface $em;
+
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     */
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @param string $searchText
+     * @param string $locale
+     * @param int $domainId
+     * @param int $offset
+     * @param int $limit
+     * @return \Shopsys\FrameworkBundle\Model\Category\Category[]
+     */
+    public function getVisibleCategoriesBySearchText(string $searchText, string $locale, int $domainId, int $offset, int $limit): array
```

Comment:
Reason for a new method instead of reuse some from CategoryRepository is because you need `offset` instead of `page`?

I'm thinking whether it could be somehow unified so I'm not forced to override behavior in two places in my project...

Reply:
Yeas that is the reason